src/main/scala/mesosphere/marathon/state/GroupManager.scala

```diff
@@ -202,10 +202,11 @@ class GroupManager @Inject() (
       }
   }
 
-  //scalastyle:off method.length
+  //scalastyle:off method.length cyclomatic.complexity
   private[state] def assignDynamicServicePorts(from: Group, to: Group): Group = {
     val portRange = Range(config.localPortMin(), config.localPortMax())
-    var taken = from.transitiveApps.flatMap(_.portNumbers) ++ to.transitiveApps.flatMap(_.portNumbers)
+    var taken = (from.transitiveApps.flatMap(app => app.hostPorts.flatten ++ app.servicePorts) ++
```

Comment:
I'm seeing a theme here where you want to use service ports instead of host ports everywhere. I don't understand it. I'm probably missing something obvious. My understanding of `portNumbers` was that they were cousins of `hostPorts`. `servicePorts` in my mind are completely different